src/modules/relayer/relayer.service.ts

```diff
@@ -14,53 +15,72 @@ import { logger } from "../../libs/logger.service";
 import { signTransaction } from "../../libs/signer.service";
 
 export async function handleExecute(address: string, transaction: Transaction) {
-  logger.info("Received execute request");
+  logger.info(`📥 Received execute request for Universal Profile ${address}`);
 
   transactionGate();
 
+  const { signature, nonce, abi, validityTimestamps } = transaction;
+
   const provider = getProvider();
 
   const universalProfile = UniversalProfile__factory.connect(address, provider);
   const keyManagerAddress = await universalProfile.owner();
-
   const keyManager = LSP6KeyManager__factory.connect(
     keyManagerAddress,
     provider
   );
 
-  const gasLimit = await keyManager.estimateGas.executeRelayCall(
-    transaction.signature,
-    transaction.nonce,
-    transaction.validityTimestamps || 0,
-    transaction.abi
-  );
+  let gasLimit: BigNumber;
+  try {
+    gasLimit = await keyManager.estimateGas.executeRelayCall(
+      signature,
+      nonce,
+      validityTimestamps || 0,
+      abi
+    );
+  } catch (error) {
+    gasLimit = BigNumber.from(3000000);
+    logger.info(
+      "⏭️ Unable to estimate gas. Setting default value to gas Limit"
+    );
+  }
 
-  const transactionData = keyManager.interface.encodeFunctionData(
-    "executeRelayCall",
-    [
-      transaction.signature,
-      transaction.nonce,
-      transaction.validityTimestamps || 0,
-      transaction.abi,
-    ]
-  );
+  const lsp6Interface = LSP6KeyManagerInit__factory.createInterface();
+
+  const transactionData = lsp6Interface.encodeFunctionData("executeRelayCall", [
+    signature,
+    nonce,
```

Comment:
in this case its more clear to keep the transaction object so developers can follow it and dont get confused by the different type of nonces